syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -595,15 +649,38 @@ def _generate_remaining_candidates(self) -> List[dict]:
         return remaining_candidates
 
     def _generate_all_candidates_on_grid(self) -> List[dict]:
-        # Get hp values that are specified as Domain
+        '''
+        This function generates all configurations on grid.
+        Returns:
+            The set of all configurations on grid.
+        '''
         hp_keys = []
         hp_values = []
-        for key, hp_range in reversed(list(self.config_space.items())):
-            if isinstance(hp_range, Domain):
-                hp_keys.append(key)
+        ## adding categorical, finiteRange, scalar parameters
+        for hp, hp_range in reversed(list(self.config_space.items())):
+            if isinstance(hp_range, Float) or isinstance(hp_range, Integer):
+                continue
+            if isinstance(hp_range, Categorical):
+                hp_keys.append(hp)
                 hp_values.append(hp_range.categories)
+            elif isinstance(hp_range, FiniteRange):
+                hp_keys.append(hp)
+                hp_values.append(hp_range.values)
+            else: # isinstance(hp_range, Hyperparameter)
+                hp_keys.append(hp)
+                hp_values.append([hp_range])
+
+        ## adding float, integer parameters
+        for hpr in self._hp_ranges._hp_ranges:
+            if hpr.name in hp_keys: continue
+            _hpr_nsamples = self.num_samples[hpr.name]
+            _normalized_points = [(idx + 0.5) / _hpr_nsamples for idx in range(_hpr_nsamples)]
+            _hpr_points = [hpr.from_ndarray(np.array([point])) for point in _normalized_points]
```

Comment:
Just to be very safe, I'd check whether there are duplicates in case of Integer type, and if so, remove them.